main.go

```diff
@@ -6,15 +6,26 @@ import (
 	"fmt"
 	"go/build"
 	"os"
+	"strings"
 
 	semver "github.com/Masterminds/semver/v3"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
 
 	"github.com/pulumi/upgrade-provider/colorize"
 	"github.com/pulumi/upgrade-provider/upgrade"
 )
 
+var (
+	// The name of our config file, without the file extension because viper supports many different config file languages.
+	defaultConfigFilename = "upgrade-config"
```

Comment:
I thought we discussed that this should start with a `.` so it is hidden.

Also, why is it `defaultConfigName`? Can you override the config name? If so, where?